consensus/spos/bls/proxy/subroundsHandler.go

```diff
@@ -0,0 +1,214 @@
+package proxy
+
+import (
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-core-go/data"
+	logger "github.com/multiversx/mx-chain-logger-go"
+
+	"github.com/multiversx/mx-chain-go/common"
+	"github.com/multiversx/mx-chain-go/consensus"
+	"github.com/multiversx/mx-chain-go/consensus/spos"
+	v1 "github.com/multiversx/mx-chain-go/consensus/spos/bls/v1"
+	v2 "github.com/multiversx/mx-chain-go/consensus/spos/bls/v2"
+	"github.com/multiversx/mx-chain-go/factory"
+	"github.com/multiversx/mx-chain-go/outport"
+)
+
+var log = logger.GetOrCreate("consensus/spos/bls/proxy")
+
+// SubroundsHandlerArgs struct contains the needed data for the SubroundsHandler
+type SubroundsHandlerArgs struct {
+	Chronology           consensus.ChronologyHandler
+	ConsensusCoreHandler spos.ConsensusCoreHandler
+	ConsensusState       spos.ConsensusStateHandler
+	Worker               factory.ConsensusWorker
+	SignatureThrottler   core.Throttler
+	AppStatusHandler     core.AppStatusHandler
+	OutportHandler       outport.OutportHandler
+	SentSignatureTracker spos.SentSignaturesTracker
+	EnableEpochsHandler  core.EnableEpochsHandler
+	ChainID              []byte
+	CurrentPid           core.PeerID
+}
+
+// subroundsFactory defines the methods needed to generate the subrounds
+type subroundsFactory interface {
+	GenerateSubrounds() error
+	SetOutportHandler(driver outport.OutportHandler)
+	IsInterfaceNil() bool
+}
+
+type consensusStateMachineType int
+
+// SubroundsHandler struct contains the needed data for the SubroundsHandler
+type SubroundsHandler struct {
```

Comment:
does this needs to be exported?

Reply:
I think we should start using exported types for constructors returns, even if we keep the internal fields unexported.
It would make it much simpler to work with these types